### PR TITLE
Feat/cards redesign

### LINKED ISSUE #1432: [2.1] New Listing card that shows both minting and listings data
[All cards cases here](https://www.notion.so/Unified-Markets-65195514608c429fbba4e0d0884fde3c)

Contemplate all the card cases descried above.

This will need the data from the new backend endpoint. Either use mocked data or wait until the endpoint is developed.

### LINKED ISSUE #1434: [2.3] Remove the assetType filter and the top bar component that toggles between primary and secondary markets
This component won’t be needed once we have all unified. Let’s leave this one to the end, once we have the cards already unified and the concepts of assetType is deprecated

### PR DESCRIPTION
**Pull request by flobarreto**
closes #1432
closes #1434 

https://user-images.githubusercontent.com/32873485/233206205-e796eeaa-3ad8-4ba3-9fd9-24bd064f3bdc.mov

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -7,5 +7,7 @@
     "type": "git",
     "url": "git+https://github.com/decentraland/marketplace.git"
   },
-  "dependencies": {}
+  "dependencies": {
```
[the remark this answers is not in the document]
Comment (juanmahidalgo) [reply package.json:10]: same goes for the `-lock` file from above.

File: webapp/src/components/AssetCard/AssetCard.tsx [hunk 2]
```diff
@@ -24,6 +28,8 @@ import { EmoteTags } from './EmoteTags'
 import { ENSTags } from './ENSTags'
 import { Props } from './AssetCard.types'
 import './AssetCard.css'
+import { LinkedProfile } from '../LinkedProfile'
+import mintingIcon from '../../images/minting.png'
```
Comment (juanmahidalgo) [inline webapp/src/components/AssetCard/AssetCard.tsx:32]: please move these imports a few lines above

File: webapp/src/components/AssetCard/AssetCard.tsx [hunk 3]
```diff
@@ -99,6 +105,104 @@ const AssetCard = (props: Props) => {
...
+                  `- ${formatWeiMANA(asset.maxListingPrice)}`}
+              </span>
+            ) : null
+        }
+      } else {
+        information = {
+          action: t('asset_card.not_for_sale'),
+          actionIcon: null,
+          price: null,
+          extraInformation: `${asset.owners} Owners`
```
Comment (juanmahidalgo) [inline webapp/src/components/AssetCard/AssetCard.tsx:168]: we need a translation for `Owners`

File: webapp/src/components/AssetCard/AssetCard.tsx [hunk 4]
```diff
@@ -99,6 +105,104 @@ const AssetCard = (props: Props) => {
...
+      }
+    }
+    return information ? (
+      <div className="CatalogItemInformation">
+        <span>
+          {information.action} &nbsp;
+          {information.actionIcon && (
+            <img
+              src={information.actionIcon}
+              alt="mint"
```
Comment (juanmahidalgo) [inline webapp/src/components/AssetCard/AssetCard.tsx:179]: is this going to be always `mint`?

File: webapp/src/components/AssetCard/AssetCard.tsx [hunk 5]
```diff
@@ -99,6 +105,104 @@ const AssetCard = (props: Props) => {
...
+              style={{ height: 14 }}
```
Comment (juanmahidalgo) [inline webapp/src/components/AssetCard/AssetCard.tsx:180]: can we set this with css?

File: webapp/src/components/AssetCard/AssetCard.tsx [hunk 6]
```diff
@@ -123,8 +228,17 @@ const AssetCard = (props: Props) => {
...
       <Card.Content>
         <Card.Header>
-          <div className="title">{title}</div>
-          {price ? (
+          <div className="title">
+            {title}
+            {isCatalogItem(asset) && (
+              <LinkedProfile
+                address={asset.contractAddress}
+                textOnly={true}
```
Comment (juanmahidalgo) [inline webapp/src/components/AssetCard/AssetCard.tsx:236]:
```suggestion
                textOnly
```

File: webapp/src/components/AssetCard/AssetCard.tsx [hunk 7]
```diff
@@ -123,8 +228,17 @@ const AssetCard = (props: Props) => {
...
+                className="Creator"
```
Comment (juanmahidalgo) [inline webapp/src/components/AssetCard/AssetCard.tsx:237]:
```suggestion
                className="creator"
```
should we use the Camelcase for classes? I think we're using other convention

File: webapp/src/components/AssetList/AssetList.tsx [hunk 8]
```diff
@@ -1,36 +1,29 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
```
Comment (juanmahidalgo) [inline webapp/src/components/AssetList/AssetList.tsx:1]: can we remove this please? 🙏

File: webapp/src/components/AssetList/AssetList.tsx [hunk 9]
```diff
@@ -1,36 +1,29 @@
...
 import { getAnalytics } from 'decentraland-dapps/dist/modules/analytics/utils'
-import { getCategoryFromSection } from '../../modules/routing/search'
 import { getMaxQuerySize, MAX_PAGE, PAGE_SIZE } from '../../modules/vendor/api'
-import { AssetType } from '../../modules/asset/types'
-import { NFT } from '../../modules/nft/types'
 import { AssetCard } from '../AssetCard'
 import { Props } from './AssetList.types'
 import './AssetList.css'
+import { getCategoryFromSection } from '../../modules/routing/search'
+import { NFTCategory } from '@dcl/schemas'
```
Comment (juanmahidalgo) [inline webapp/src/components/AssetList/AssetList.tsx:11]: please move this import above 🙏

File: webapp/src/components/AssetList/AssetList.tsx [hunk 10]
```diff
@@ -66,7 +60,14 @@ const AssetList = (props: Props) => {
       }
     }
     return 'nft_list.simple_empty'
-  }, [assets.length, search, section, isManager])
+  }, [catalogItems.length, search, section, isManager])
+
+  console.log(
```
Comment (juanmahidalgo) [inline webapp/src/components/AssetList/AssetList.tsx:65]: 👀

File: webapp/src/components/AssetList/AssetList.tsx [hunk 11]
```diff
@@ -77,62 +78,20 @@ const AssetList = (props: Props) => {
...
-        {assets.length > 0
-          ? assets.map((assets, index) => (
-              <AssetCard
-                isManager={isManager}
-                key={assetType + '-' + assets.id + '-' + index}
-                asset={assets}
-              />
+        {catalogItems.length > 0
+          ? catalogItems.map((catalogItem, index) => (
+              <>
```
Comment (juanmahidalgo) [inline webapp/src/components/AssetList/AssetList.tsx:83]: I believe this fragment is not needed

File: webapp/src/components/AssetList/AssetList.tsx [hunk 12]
```diff
@@ -77,62 +78,20 @@ const AssetList = (props: Props) => {
...
       ) : null}
       <Card.Group>
-        {assets.length > 0
-          ? assets.map((assets, index) => (
-              <AssetCard
-                isManager={isManager}
-                key={assetType + '-' + assets.id + '-' + index}
-                asset={assets}
-              />
+        {catalogItems.length > 0
```
Comment (juanmahidalgo) [inline webapp/src/components/AssetList/AssetList.tsx:81]: I believe we don't need this check, we can just call the `.map` and if it's empty it won't do anything

File: webapp/src/components/AssetList/AssetList.types.ts [hunk 13]
```diff
@@ -10,14 +7,11 @@ import {
   ClearFiltersAction
 } from '../../modules/routing/actions'
 import { Section } from '../../modules/vendor/routing/types'
-import { AssetType } from '../../modules/asset/types'
+import { CatalogItem } from '@dcl/schemas'
```
Comment (juanmahidalgo) [inline webapp/src/components/AssetList/AssetList.types.ts:10]: move this import up please 🙏

File: webapp/src/components/AssetPage/SaleActionBox/ItemSaleActions/ItemSaleActions.types.ts [hunk 14]
```diff
@@ -1,8 +1,8 @@
-import { Item } from '@dcl/schemas'
+import { Item, CatalogItem } from '@dcl/schemas'
 import { Wallet } from 'decentraland-dapps/dist/modules/wallet/types'
 
 export type Props = {
-  item: Item
+  item: Item | CatalogItem
```
Comment (juanmahidalgo) [inline webapp/src/components/AssetPage/SaleActionBox/ItemSaleActions/ItemSaleActions.types.ts:5]: Do we need this change?

File: webapp/src/components/OnSaleOrRentList/AssetCell/AssetCell.tsx [hunk 15]
```diff
@@ -25,7 +25,7 @@ const AssetCell = ({ asset, link: linkProp }: Props) => {
     ? linkProp
     : 'tokenId' in asset
     ? locations.nft(asset.contractAddress, asset.tokenId)
-    : locations.item(asset.contractAddress, asset.itemId!)
+    : locations.item(asset.contractAddress, asset.id!)
```
Comment (juanmahidalgo) [inline webapp/src/components/OnSaleOrRentList/AssetCell/AssetCell.tsx:28]: is this correct?

File: webapp/src/config/env/dev.json [hunk 16]
```diff
@@ -1,7 +1,7 @@
 {
   "NETWORK": "goerli",
   "CHAIN_ID": "5",
-  "NFT_SERVER_URL": "https://nft-api.decentraland.zone/v1",
+  "NFT_SERVER_URL": "http://localhost:5001/v1",
```
Comment (juanmahidalgo) [inline webapp/src/config/env/dev.json:4]: please revert this 🙈

File: webapp/src/modules/asset/utils.ts [hunk 17]
```diff
@@ -64,6 +64,9 @@ export function getAssetUrl(asset: Asset, isManager?: boolean) {
   if ('itemId' in asset && asset.itemId !== null) {
     return locations.item(asset.contractAddress, asset.itemId)
   }
+  if ('id' in asset && asset.itemId !== null) {
```
Comment (juanmahidalgo) [inline webapp/src/modules/asset/utils.ts:67]: now that we have `itemId` we don't need this right?

File: webapp/src/modules/catalog/sagas.ts [hunk 18]
```diff
@@ -0,0 +1,35 @@
...
+function* handleFetchCatalogRequest(action: FetchCatalogRequestAction) {
+  const { filters } = action.payload
+  try {
+    const { data, total }: { data: CatalogItem[]; total: number } = yield call(
+      [catalogAPI, 'fetch'],
+      filters
+    )
+    yield put(fetchCatalogSuccess(data, total, filters))
+  } catch (error) {
+    console.log('error: ', error)
```
Comment (juanmahidalgo) [inline webapp/src/modules/catalog/sagas.ts:27]: let's remove this one

File: webapp/src/modules/transak/sagas.ts [hunk 19]
```diff
@@ -29,7 +29,7 @@ function* handleOpenTransak(action: OpenTransakAction) {
       appCluster: config.get('TRANSAK_PUSHER_APP_CLUSTER')
     }
   }
-  const tokenId = isNFT(asset) ? asset.tokenId : asset.itemId
+  const tokenId = isNFT(asset) ? asset.tokenId : asset.id
```
Comment (juanmahidalgo) [inline webapp/src/modules/transak/sagas.ts:32]: do we need to revert this change now that we have `itemId`?

File: webapp/src/components/AssetCard/AssetCard.tsx (hunk 4 above)
Comment (flobarreto) [reply webapp/src/components/AssetCard/AssetCard.tsx:179]: yes

File: webapp/src/components/AssetPage/BestBuyingOption/BestBuyingOption.tsx [hunk 20]
```diff
@@ -27,6 +28,7 @@ import { BuyNFTButtons } from '../SaleActionBox/BuyNFTButtons'
 import { BelowTabs } from '../ListingsTableContainer/ListingsTableContainer.types'
 import { BuyOptions, Props } from './BestBuyingOption.types'
 import styles from './BestBuyingOption.module.css'
+import { ItemSaleActions } from '../SaleActionBox/ItemSaleActions'
```
Comment (juanmahidalgo) [inline webapp/src/components/AssetPage/BestBuyingOption/BestBuyingOption.tsx:31]: please move this import up 🙏

File: webapp/src/components/AssetPage/BestBuyingOption/BestBuyingOption.tsx [hunk 21]
```diff
@@ -63,8 +65,12 @@ const BestBuyingOption = ({ asset, tableRef }: Props) => {
         }
         const sortBy = OrderSortBy.CHEAPEST
 
-        if (asset.network === Network.MATIC && asset.itemId) {
-          params.itemId = asset.itemId
+        if (asset.network === Network.MATIC) {
+          if (isCatalogItem(asset)) {
```
Comment (juanmahidalgo) [inline webapp/src/components/AssetPage/BestBuyingOption/BestBuyingOption.tsx:69]: do we still need this? now we have `itemId` as part of the asset

File: webapp/src/components/AssetPage/SaleActionBox/ItemSaleActions/ItemSaleActions.container.ts [hunk 22]
```diff
@@ -2,10 +2,15 @@ import { connect } from 'react-redux'
...
+import ItemSaleActions from './ItemSaleActions'
 
-const mapState = (state: RootState): MapStateProps => ({
-  wallet: getWallet(state)
-})
+const mapState = (state: RootState): MapStateProps => {
+  const wallet = getWallet(state)
 
-export default connect(mapState)(SaleRentActionBox)
+  console.log('wallet', wallet)
```
Comment (juanmahidalgo) [inline webapp/src/components/AssetPage/SaleActionBox/ItemSaleActions/ItemSaleActions.container.ts:10]: remove console.log :)

File: webapp/src/lib/mana.ts [hunk 23]
```diff
@@ -22,6 +23,10 @@ export function formatWeiMANA(
     return getMinimumValueForFractionDigits(maximumFractionDigits).toString()
   }
 
+  if (formatMillion && value > 1000000) {
```
Comment (juanmahidalgo) [inline webapp/src/lib/mana.ts:26]: can remove this now that we have the other util method in `webapp/src/components/AssetCard/utils.tsx` ?

File: webapp/src/modules/catalog/selectors.ts [hunk 24]
```diff
@@ -0,0 +1,27 @@
...
+import { createSelector } from 'reselect'
+import { RootState } from '../reducer'
+import { FETCH_CATALOG_REQUEST } from './actions'
+
+export const getState = (state: RootState) => state.catalogItem
+export const getData = (state: RootState) => getState(state).data
+export const getError = (state: RootState) => getState(state).error
+export const getLoading = (state: RootState) => getState(state).loading
+
+export const isFetchingCatalogItems = (
```
Comment (juanmahidalgo) [inline webapp/src/modules/catalog/selectors.ts:11]: Are we using this method?? If not, let's remove it :)

File: webapp/src/modules/catalog/reducer.ts [hunk 25]
```diff
@@ -0,0 +1,69 @@
...
+  loading: [],
+  error: null
+}
+
+type CatalogReducerAction =
+  | FetchCatalogFailureAction
+  | FetchCatalogRequestAction
+  | FetchCatalogSuccessAction
+
+export function catalogItemReducer(
```
Comment (juanmahidalgo) [inline webapp/src/modules/catalog/reducer.ts:32]:
```suggestion
export function catalogReducer(
```
Can we remove the `item` from the name? I think this one looks more accurate, wdyt?

**Review by juanmahidalgo — APPROVED** [review]
🚀🚀🚀
